code/native/mpris.py

```diff
@@ -344,16 +354,17 @@ class StreamKeysMPRIS(MediaPlayer):
 
     def __setattr__(self, name, value):
         """Override to set the DBus object properties and emit signals."""
-        if name == "Volume":
-            value = max(0.0, min(1.0, value))
-            value = round(value, 2)
-            send_msg(Message(command=Command.VOLUME, args=value))
         super().__setattr__(name, value)
 
     def __init__(self):
         super().__init__(name="streamkeys",
                          identity="Chrome StreamKeys extension")
 
+    @Player.Volume.setter
+    def Volume(self, value):
+        Player.Volume.fset(self, value)
+        send_msg(Message(command=Command.VOLUME, args=value))
```

Comment:
Since `Player.Volume.fset()` will modify `value` to set it in the 0.0-1.0 range, let's not send the old `value` in `args`. Instead we can use `args=self._volume`.